jbpm-console-ng-process-runtime/jbpm-console-ng-process-runtime-client/src/main/java/org/jbpm/console/ng/pr/client/editors/instance/list/variables/dash/DataSetProcessInstanceWithVariablesListViewImpl.java

```diff
@@ -372,23 +373,57 @@ public String getValue( ProcessInstanceSummary object ) {
     }
 
     public void initExtraButtons( final ExtendedPagedTable<ProcessInstanceSummary> extendedPagedTable ) {
+        final Button selectAllButton = GWT.create(Button.class);
+        selectAllButton.setText(constants.SelectAll());
+        selectAllButton.setIcon(IconType.SQUARE_O);
+        selectAllButton.setTitle(constants.SelectAllTooltip());
+        selectAllButton.setIcon(IconType.SQUARE_O);
+        selectAllButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                selectedProcessInstances.clear();
+                if (IconType.SQUARE_O.equals(selectAllButton.getIcon())) {
+                    selectedProcessInstances.addAll(presenter.getDisplayedProcessInstances());
+                    presenter.refreshGrid();
+                    selectAllButton.setIcon(IconType.CHECK_SQUARE_O);
+                    selectAllButton.setText(constants.UnselectAll());
+                    selectAllButton.setTitle(constants.UnselectAllTooltip());
+                } else {
+                    selectAllButton.setIcon(IconType.SQUARE_O);
+                    selectAllButton.setText(constants.SelectAll());
+                    selectAllButton.setTitle(constants.SelectAllTooltip());
+                }
+                presenter.refreshGrid();
+                controlBulkOperations();
+            }
+        });
+        extendedPagedTable.getLeftToolbar().clear();
+        extendedPagedTable.getLeftToolbar().add(selectAllButton);
```

Comment:
I think it would actually be handy if this button is placed in the right, just before the bulk actions. I'm imagining something like select all and notify, that way both actions are close together. What do you think @nmirasch?